plugins/sinks/file/file.go

```diff
@@ -65,10 +66,13 @@ func (s *Sink) Init(ctx context.Context, config map[string]interface{}) (err err
 }
 
 func (s *Sink) Sink(ctx context.Context, batch []models.Record) (err error) {
-	var data []models.Metadata
```

Comment:
What if, instead of having a global variable, we do collect as it is and write in this Sink (not in the Close part).

We can write in this step by Opening an `append-only` or `write-only` file like [this](https://yourbasic.org/golang/append-to-file/).

Reply:
hey, @mabdh actually the issue here is with the output format being `json` we cannot append data to the file directly, hence we discussed keeping it in a way that replaces the file instead of making a config for `append`. Thus we maintain a variable to store the data and save it to the file only once at `Close()`.

Reply:
For json, actually we can if we use http://ndjson.org/ like format

Reply:
Yes, sure thing I will look more into it actually. @mabdh

Reply:
@mabdh have updated with `ndjson` format and seems to work fine to me. Working to make the file append/replace configurable for user now. Let me know if there are any concerns